### PR TITLE
feat(components): [input-number] support slot custom icons

### PR DESCRIPTION
**Pull request by selicens**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

| native | use Slots |
| ------ | ---------- |
| image |  image |

### CODE REVIEW

File: docs/en-US/component/input-number.md [hunk 1]
```diff
@@ -79,6 +79,14 @@ input-number/controlled
 
 :::
 
+## Custom Icon
```
Comment (btea) [inline docs/en-US/component/input-number.md:82]:
```suggestion
## Custom Icon ^(2.6.3)
```

File: docs/examples/input-number/custom.vue [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      <el-icon>
+        <Minus />
+      </el-icon>
+    </template>
+    <template #append>
+      <el-icon>
+        <Plus />
+      </el-icon>
+    </template>
+  </el-input-number>
```
Comment (btea) [inline docs/examples/input-number/custom.vue:28]: It seems that the layout here can also be adjusted, it currently looks too compact.

Comment (selicens) [reply docs/examples/input-number/custom.vue]: image

File: docs/en-US/component/input-number.md [hunk 3]
```diff
@@ -103,6 +111,13 @@ input-number/controlled
 | value-on-clear ^(2.2.0) | value should be set when input box is cleared    | ^[number] / ^[null] / ^[enum]`'min' \| 'max'` | —         |
 | validate-event          | whether to trigger form validation               | ^[boolean]                                    | true      |
 
+### Slots
+
+| Name    | Description                  |
+| ------- | ---------------------------- |
+| prepend | custom input box front icon  |
```
Comment (btea) [inline docs/en-US/component/input-number.md:118]:
```suggestion
| prepend ^(2.6.3) | custom input box front icon  |
```

File: docs/en-US/component/input-number.md [hunk 4]
```diff
@@ -103,6 +111,13 @@ input-number/controlled
...
+| append  | custom input box post icon   |
```
Comment (btea) [inline docs/en-US/component/input-number.md:119]: Same as above.

File: docs/en-US/component/input-number.md (hunk 3 above)
Comment (kooriookami) [reply docs/en-US/component/input-number.md:118]: Do you think `decrease` is better then `prepend`?

File: docs/en-US/component/input-number.md (hunk 4 above)
Comment (kooriookami) [reply docs/en-US/component/input-number.md:119]: May be `increase` is good.

File: docs/en-US/component/input-number.md (hunk 3 above)
Comment (selicens) [reply docs/en-US/component/input-number.md:118]: > Do you think is better then ?`decrease``prepend`

It's not impossible either, it's just that following the naming conventions of similar components would be more convenient to remember

Comment (kooriookami) [reply docs/en-US/component/input-number.md:118]: I know you refered to el-input. But the function is not the same. The icon slot in el-input-number only used for replacing the decrease and increase icon, It's not universal. @btea What do you think?
image

Comment (btea) [reply docs/en-US/component/input-number.md]: It makes sense. In fact, there are cases where these two icons are on the right side at the same time. In this case, the name suggested by @kooriookami  seems clearer.

Comment (btea) [reply docs/en-US/component/input-number.md]: Maybe it can be clearer, such as `increase-icon`/`decrease-icon`.

Comment (selicens) [reply docs/en-US/component/input-number.md]: okay

File: packages/components/input-number/src/input-number.vue [hunk 5]
```diff
@@ -17,10 +17,12 @@
       :class="[ns.e('decrease'), ns.is('disabled', minDisabled)]"
       @keydown.enter="decrease"
     >
-      <el-icon>
-        <arrow-down v-if="controlsAtRight" />
-        <minus v-else />
-      </el-icon>
+      <slot name="decreaseIcon">
```
Comment (btea) [inline packages/components/input-number/src/input-number.vue:20]:
```suggestion
      <slot name="decrease-icon">
```
Use hyphens to be consistent with other components.

Comment (selicens) [reply packages/components/input-number/src/input-number.vue:20]: Thanks to Vue's handling, this does not affect the results

Comment (btea) [reply packages/components/input-number/src/input-number.vue:20]: Yes. It’s just that I think it might be better to keep the component styles consistent.

Comment (selicens) [reply packages/components/input-number/src/input-number.vue:20]: okay

**Review by btea — APPROVED** [review]
LGTM 👍

**Review by kooriookami — APPROVED** [review]

**Pull request merged**